edit-post/components/plugin-more-menu-item/index.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { compose } from '@wordpress/element';
+import { Slot, Fill, withContext, MenuItemsGroup } from '@wordpress/components';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import MoreMenuItemLayout from './more-menu-item-layout';
+
+/**
+ * Name of slot in which the sidebar should fill.
+ *
+ * @type {String}
+ */
+const SLOT_NAME = 'PluginMoreMenuItem';
+
+function PluginMoreMenuItem( { title, onClick, icon, isActive } ) {
+	return (
+		<Fill name={ SLOT_NAME }>
+			{ ( props ) => {
+				return (
+					<MoreMenuItemLayout
+						isActive={ isActive }
+						title={ title }
+						onClick={ () => {
+							onClick();
+							props.onClose();
+						} }
+						icon={ icon } />
+				);
+			} }
+		</Fill>
+	);
+}
+
+PluginMoreMenuItem = compose( [
+	withContext( 'pluginName' )( ( pluginName, { target } ) => {
+		return {
+			target: `${ pluginName }/${ target }`,
+		};
+	} ),
+	withSelect( ( select, { target } ) => ( {
+		isActive: select( 'core/edit-post' ).getActiveGeneralSidebarName() === target,
+	} ) ),
+	withDispatch( ( dispatch, { type, target, isActive } ) => {
+		let onClick = noop;
+		if ( isActive ) {
+			onClick = dispatch( 'core/edit-post' ).closeGeneralSidebar;
+		} else {
+			switch ( type ) {
+				case 'sidebar':
```

Comment:
In here this code checks the type, but it doesn't do the same for `isActive` set to truthy value.

In case there is `type` set to the value other than `sidebar` this code will override `onClick` passed as prop with `noop when `isActive` is falsy making this item useless.